src/utils/SortingNetworkGenerator.java

```diff
@@ -99,6 +116,10 @@ public static String encodeNetworkAndDisplay(String name, Integer[] indices, int
             indicesInt[i] = indices[i];
         }
         if (!encodeNetwork(indicesInt, path)) {
+			if (indicesInt.length == 0) {
+				JOptionPane.showMessageDialog(null, "Sort does not compare indices; An empty sorting network cannot be generated.",
+					"File not saved", JOptionPane.ERROR_MESSAGE);
+			}
```

Comment:
Instead of checking for the error in creating the sorting network here, why not show this message dialog in the `encodeNetwork` function itself? (also anyway this code still uses the compare-to-0)